inferentia/scripts/gen_triton_model.py

```diff
@@ -207,29 +153,46 @@ def initialize(self, args):
         self.model_config = model_config = json.loads(args['model_config'])
 
         self.input_dict = {}
+        expected_input_count = 0
         for config_input in model_config['input']:
-            self.input_dict[config_input['name']] = [
-                config_input['data_type'], config_input['dims']
+            index = self._validate_and_get_index(config_input['name'])
+            self.input_dict[index] = [
```

Comment:
Do you need a dict if index is used as key? List seems to be sufficient.

Reply:
We need to handle cases like INPUT__0 and INPUT__2, where INPUT__1 is missing. Using a dict helps in detecting missing indices with minimal iterations across input lists.